src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/authorization/ReferralAccessChecker.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization
+
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+
+class ReferralAccessChecker(
+  private val userTypeChecker: UserTypeChecker,
+  private val eligibleProviderMapper: EligibleProviderMapper,
+  private val userAccessScopeMapper: UserAccessScopeMapper,
+) {
+  fun forServiceProviderUser(referral: Referral, user: AuthUser): Boolean {
```

Comment:
Is this class/function used somewhere yet? I couldn't find it

Reply:
no it's not - yet. i mention this in the PR description. andrew and ryan are working on integration tests to allow us to roll this out across the controllers.